client/rpc_sender.go

```diff
@@ -109,6 +111,9 @@ func (s *rpcSender) Send(_ context.Context, call proto.Call) {
 		break
 	}
 	if err != nil {
-		call.Reply.Header().SetGoError(err)
+		return nil, proto.NewError(err)
 	}
+	pErr := br.Error
```

Comment:
yep, though we can make it invisible to everything else if we move it into a message that holds a batch request and an error. It's one of the things I'm going to consider when this stuff here is through.